### DISCUSSION TITLE
Allow multiple labels on an element (or at least document that this isn't currently allowed.)

### DISCUSSION THREAD

**Discussion by TheFeshy** [typst/typst#2457] [General]
### Description

Consider the following: 

```
#show <a>: set text(blue)
#show <b>: set text(weight:"bold")

Lorem(2)<a><b>
```

I would have expected the text to be both bold and blue - however, the text is only bolded.  I suspect the label <a> is overwritten by label <b>.

There is nothing about this limitation in the documents, and as a result it took some time to track down what was happening.  My use case was more complex than this.

It can be worked around in a number of ways, from using other methods of styling to something as simple as

`box()[Lorem(2)<a>]<b>`

### Use Case

The specific scenario I was attempting to use this was to create pages within my document that lacked certain attributes - for instance, the background or the footer.  To do this, I was placing tags in the page such as <no-background> and <no-footer>, and the corresponding page background/footer functions did a query for those terms and skipped drawing them if found.  However, if the labels were back-to-back, only the second would be detected by the query.

But it would also be useful, conceptually, to tag elements with styling they might need without regard to overlap (e.g. if there are elements in the union between elements that need to be bold and elements that should be blue.)

**Comment by emilyyyylime** [comment]
your code didn't get formatted correctly and I think that messed up your ability to get your point across — I don't understand what the issue / feature request you're referring to is our how the code is relevant to it. Please correct this

**Comment by Enivex** [comment]
I copy pasted your code, and got blue, bold text

![image](https://github.com/typst/typst/assets/2191754/b5cf804d-f477-4ca7-8fba-7b304adebdfe)

**Comment by MDLC01** [comment]
I think this is what they meant:
```typst
#show <a>: set text(blue)
#show <b>: set text(weight: "bold")

#lorem(2) <a> <b>
```
which renders as
!["Lorem ipsum." in bold.](https://github.com/typst/typst/assets/57839069/7762c835-86b5-40b4-a6ea-47f4789caeba)

**Reply by TheFeshy** [reply]
Yes, this is what I had.  Sorry, I didn't realize the formatting ate the tags.  Thank you very much for the correction!

**Comment by laurmaedje** [comment]
Generally speaking, labels are more intended to identify specific elements rather than styling classes of related elements. For the latter, it's recommended to define a semantically named function that wraps the relevant element. Functions are more flexible and better encapsulated than labels.

I know that the label documentation states styling as an explicit use case. It should likely be updated.

**Comment by minijackson** [comment]
Hello! I'm also interested in having multiple labels on an element.

I'm trying to build a Sphinx extension that builds PDFs through Typst, and Sphinx often has multiple targets / labels that can apply on a elements.

For example, on a file name `something.rst`:

```rst
.. _label:

Title
=====
```

You can refer to "Title" with either `` :ref:`label` `` or `` :doc:`something` ``. Having multiple labels would simplify it a lot.

**Comment by kudrykv** [comment]
I have a case where it's easier for me to build multiple labels for an element rather than figuring how to make a "definitely unique" one. For that, I need to label the same content with multiple labels.

I could got away with it using `#hide` (just in case) and `~`:
```typst
My heading #hide[~#label("one") ~#label("two") ~#label("three")]
```

**Reply by emilyyyylime** [reply]
FYI, you should also use `#place()` to make sure the hidden content won't affect layout. An arguably better approach would be to replace the non-breaking spaces with `#metadata` objects.

**Comment by Jollywatt** [comment]
Another use case for multiple labels:

```typ

#show <eq>: set math.equation(numbering: "(1)")

This equation is unnumbered:
$
  1 + 1 = 2
$
but this one should be:
$
  a^2 + b^2 = c^2
$ <eq> <pythag>
See @pythag.
```

**Reply by Enivex** [reply]
Iirc labels aren't intended to be used for styling.

**Reply by Jollywatt** [reply]
Yeah. It's a pity, because their light syntax makes it tempting to also use them like one would use CSS classes: to add semantic information that can be used e.g., for styling, to mark things to be excluded from word counts, etc.
